### PR TITLE
Test PoseExtrapolator

### PR DESCRIPTION
**Pull request by gaschler**

### CODE REVIEW

File: cartographer/mapping/pose_extrapolator_test.cc [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+                    .angularDistance(expected_orientation),
+                kPrecision);
+  }
+  Eigen::Vector3d gravity_acceleration(1.6, 2.0, 8.0);
+  for (int i = 0; i < 10; ++i) {
+    current_time += common::FromSeconds(kGravityTimeConstant);
+    extrapolator->AddImuData(
+        sensor::ImuData{current_time, gravity_acceleration, angular_velocity});
+  }
+  {
```
Comment (pifon2a) [inline cartographer/mapping/pose_extrapolator_test.cc]: You do not have create a scope for every vector/quaternion.

File: cartographer/transform/rigid_transform_test_helpers.h [hunk 2]
```diff
@@ -40,8 +40,8 @@ Eigen::Transform<T, 3, Eigen::Affine> ToEigen(const Rigid3<T>& rigid3) {
 }
 
 MATCHER_P2(IsNearly, rigid, epsilon,
-           std::string(std::string(negation ? "isn't" : "is", " nearly ") +
-                       rigid.DebugString())) {
+           std::string(std::string(negation ? "isn't" : "is") +
```
Comment (pifon2a) [inline cartographer/transform/rigid_transform_test_helpers.h]: Some of the "std::string()" looks redundant:

Maybe: std::string(negation ? "isn't nearly" : "is nearly")  + std::string(rigid.DebugString());
Or even: std::string(negation ? "isn't nearly" : "is nearly")  + rigid.DebugString();

File: cartographer/mapping/pose_extrapolator_test.cc (hunk 1 above)
Comment (gaschler) [reply cartographer/mapping/pose_extrapolator_test.cc]: Done

File: cartographer/transform/rigid_transform_test_helpers.h (hunk 2 above)
Comment (gaschler) [reply cartographer/transform/rigid_transform_test_helpers.h]: Done (the latter)

**Comment by gaschler** [comment]
@wally-the-cartographer merge

**Comment by wally-the-cartographer** [comment]
Merge requested by authorized user gaschler. Merge queue now has a length of 1.

**Pull request merged**